src/app/manager-page/student-card/student-card.component.ts

```diff
@@ -0,0 +1,50 @@
+import { getReviewData } from '../../store/review/review.selectors';
+import { ReviewData } from '../../shared/reviewData';
+import { Store } from '@ngrx/store';
+import { Student } from '../../shared/student';
+import { Observable, of } from 'rxjs';
+import { Component, OnInit } from '@angular/core';
+import { switchMap } from 'rxjs/operators';
+import { getSelectedStudentId, getStudentById } from 'src/app/store/students/students.selectors';
+
+@Component({
+  selector: 'app-student-card',
+  templateUrl: './student-card.component.html',
+  styleUrls: ['./student-card.component.scss']
+})
+export class StudentCardComponent implements OnInit {
+
+  studentId$!: Observable<number>;
+  student!: Student;
+  reviewsData$!: Observable<ReviewData>;
+  reviewData = {} as ReviewData;
+  averages = [0, 0, 0, 0, 0];
+
+  constructor(private store: Store) { }
+
+  ngOnInit(): void {
+    this.studentId$ = this.store.select(getSelectedStudentId);
+    this.studentId$.pipe(
+      switchMap(studentId => (this.store.select(getStudentById(studentId)) as Observable<Student>))
+    ).subscribe((student) => {
+      if (student) {
+        this.student = student;
+    }});
```

Comment:
```suggestion
    this.student$ = this.store.pipe(
      select(getSelectedStudentId),
      switchMap(studentId => (this.store.select(getStudentById(studentId)) as Observable<Student>))
```

I think you can you use observable straight away :)